specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -225,11 +255,27 @@ function TreeView<SCHEMA extends AnyTree>({
     <Container.Full>
       <header className="flex items-center gap-2 overflow-x-auto sm:flex-wrap sm:overflow-x-visible">
         <TableIcon label name={table.name} />
-        <H2 title={treeDefinition.get('remarks') ?? undefined}>
-          {treeDefinition.get('name')}
-        </H2>
+        {definitions.length > 1 ? (
+          <Select
+            className="w-max"
+            value={treeDefinition.id}
+            onValueChange={(newDefinition: string): void => {
+              setNewDefinition(f.parseInt(newDefinition) as number);
+            }}
+          >
+            {definitions.map(({ id, name }) => (
+              <option key={name as string} value={id}>
+                {name as string}
+              </option>
+            ))}
+          </Select>
+        ) : (
+          <H2 title={(treeDefinition.remarks as string) ?? undefined}>
+            {treeDefinition.name as string}
+          </H2>
+        )}
         <ResourceEdit
-          resource={treeDefinition}
+          resource={deserializeResource(treeDefinition)}
```

Comment:
this will create a new object on each render, which is possible going to break ResourceEdit, or cause performance issues - put this into a useMemo?